contrib/endpoints/src/api_manager/context/request_context.cc

```diff
@@ -226,6 +226,18 @@ void RequestContext::FillCheckRequestInfo(
   info->allow_unregistered_calls = method()->allow_unregistered_calls();
 }
 
+void RequestContext::FillAllocateQuotaRequestInfo(
+    service_control::QuotaRequestInfo *info) {
+  FillOperationInfo(info);
+
+  info->client_ip = request_->GetClientIP();
+  info->method_name = this->method_call_.method_info->name();
+
+  // TODO(jaebong) need to set quota rule and metric rule
+  info->quota_rule_ = nullptr;
```

Comment:
you can assume method_info has a function 

get_metric_cost_map().   

it return map<string, int>*.

if nullptr,  or empty in size,  not need to call Quota.

Otherwise, just pass the pointer to info.

In the proto,  use that to construct quota metric lists. in the request proto